library/libwaku.nim

```diff
@@ -383,6 +384,119 @@ proc waku_relay_unsubscribe(
 
   return RET_OK
 
+proc waku_relay_get_num_connected_peers(
+    ctx: ptr Context, pubSubTopic: cstring, callback: WakuCallBack, userData: pointer
+): cint {.dynlib, exportc.} =
+  ctx[].userData = userData
+
+  let pst = pubSubTopic.alloc()
+  defer:
+    deallocShared(pst)
+
+  let numConnPeersRes = waku_thread.sendRequestToWakuThread(
+    ctx,
+    RequestType.RELAY,
+    RelayRequest.createShared(RelayMsgType.LIST_CONNECTED_PEERS, PubsubTopic($pst)),
+  )
+
+  if numConnPeersRes.isErr():
+    foreignThreadGc:
+      let msg = "Error in waku_relay_get_num_connected_peers: " & $numConnPeersRes.error
+      callback(RET_ERR, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+      return RET_ERR
+
+  let numConnPeers = numConnPeersRes.get()
+  foreignThreadGc:
+    callback(
+      RET_OK, unsafeAddr numConnPeers[0], cast[csize_t](len(numConnPeers)), userData
+    )
+
+  return RET_OK
+
+proc waku_relay_get_num_peers_in_mesh(
+    ctx: ptr Context, pubSubTopic: cstring, callback: WakuCallBack, userData: pointer
+): cint {.dynlib, exportc.} =
+  ctx[].userData = userData
+
+  let pst = pubSubTopic.alloc()
+  defer:
+    deallocShared(pst)
+
+  let numPeersInMeshRes = waku_thread.sendRequestToWakuThread(
+    ctx,
+    RequestType.RELAY,
+    RelayRequest.createShared(RelayMsgType.LIST_MESH_PEERS, PubsubTopic($pst)),
+  )
+
+  if numPeersInMeshRes.isErr():
+    foreignThreadGc:
+      let msg = "Error in waku_relay_get_num_peers_in_mesh: " & $numPeersInMeshRes.error
+      callback(RET_ERR, unsafeAddr msg[0], cast[csize_t](len(msg)), userData)
+      return RET_ERR
+
+  let numPeersInMesh = numPeersInMeshRes.get()
+  foreignThreadGc:
+    callback(
+      RET_OK, unsafeAddr numPeersInMesh[0], cast[csize_t](len(numPeersInMesh)), userData
+    )
+
+  return RET_OK
+
+proc waku_lightpush_publish(
+    ctx: ptr Context,
+    pubSubTopic: cstring,
+    jsonWakuMessage: cstring,
+    callback: WakuCallBack,
+    userData: pointer,
+): cint {.dynlib, exportc, cdecl.} =
+  ctx[].userData = userData
+
+  if isNil(callback):
+    return RET_MISSING_CALLBACK
```

Comment:
Why is this needed here but not in all other procs?

Reply:
> Why is this needed here but not in all other procs?

Good point! Because some checks are left. I will submit an enhancement in a separate PR